From ddee93d7cd5a32eca08bfd9d7c7a6835eb39164d Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.org>
Date: Fri, 20 Sep 2019 10:10:42 +0100
Subject: [PATCH 538/587] drm/v3d: Fix merge error memory leak

Cherry-picking "drm/v3d: Refactor job management." from 4.19 lost a
call to drm_sched_job_cleanup introduced in 4.20, causing a memory
leak.

Signed-off-by: Phil Elwell <phil@raspberrypi.org>
---
 drivers/gpu/drm/v3d/v3d_sched.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/v3d/v3d_sched.c b/drivers/gpu/drm/v3d/v3d_sched.c
index fceefad3f8d1..ef8e76248c1c 100644
--- a/drivers/gpu/drm/v3d/v3d_sched.c
+++ b/drivers/gpu/drm/v3d/v3d_sched.c
@@ -59,6 +59,8 @@ v3d_job_free(struct drm_sched_job *sched_job)
 {
 	struct v3d_job *job = to_v3d_job(sched_job);
 
+	drm_sched_job_cleanup(sched_job);
+
 	v3d_job_put(job);
 }
 
-- 
2.17.1

